docs/development/design/allow-slug-change-on-versions.rst

```diff
@@ -0,0 +1,79 @@
+Allow Slug Change on Versions
+=============================
+
+Currently we don't allow users to change this value.
+We generate one for them automatically based on the ``verbose_name`` of the version
+(branch name or tag name).
+
+Users care a lot about URLs,
+we use the slug to generate the URL of a version.
+Sometimes the generated slug isn't what they expect.
+
+We already modify slugs of some projects from the admin when requested,
+this document describes the benefits of allowing users to do it by themselves and implementation details.
+
+Advantages of changing the slug
+-------------------------------
+
+Users can have more control over their version's URLs.
+
+Users can make their versions compatible with :PEP:`440`,
+so they are sorted the way the user expects.
+
+Current usage of slugs
+----------------------
+
+Slugs are used heavily in our code,
+to identify a version from the API and to serve the version.
+And the slugs are shown on the version selector of the footer.
+
+Current workarrounds to change slug
+-----------------------------------
+
+We currently change the slug of some projects from the admin when requested.
+This is prone to error, since we can introduce an invalid character in the new slug.
+
+Users can rename their branches or tags,
+this isn't always posible for already released versions,
+and probably involves changing a lot of places outside the project.
+
+Problems of changing the slug
+-----------------------------
+
+Existing URLs will break.
+Although, users can create an exact redirect to fix this::
+
+   /en/{ current_slug }/$rest -> /en/{ new-slug }/
+
+Using another field as alias
+----------------------------
+
+Another way to get around this is by creating a new field as alias (as we do with subprojects).
+This option requires a lot of changes in our querysets and code related to serve docs.
+
+Another suggestion is to use the ``verbose_name`` of the version,
+this is a problem since the verbose_name of the version isn't slugified
+(it can create some problems in how we handle URLs),
+and we use the ``verbose_name`` to show the *real* name of the version (branch name or tag).
+
+Restrictions
+------------
+
+We need to check that the new slug is compatible with our current code.
+
+We can't allow users to change the slug of machine created versions,
+because those have a special meaning in our code (like ``stable`` and ``latest``).
```

Comment:
We already allow users to create a `latest` branch that isn't machine created, the behavior is like any other branch.

https://github.com/readthedocs/readthedocs.org/blob/b61663bef1056b2d1f8f225308281384970be585/readthedocs/rtd_tests/tests/test_sync_versions.py#L527-L587

Reply:
Right, but won't that break places where we expect a `latest` slug on a project?

Reply:
I don't think so, the only parts where this has a special meaning is when we do a sync for versions and when sorting. So it would be sorted first (like a machine created version), but probably that's what users want.